apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/ExplanationOfBenefitE2E.java

```diff
@@ -53,42 +59,87 @@ public void setupTest() {
-        /* RestAssured uses Groovy under the hood, so our search query closures can use Groovy's collection API.
-         * See https://www.baeldung.com/rest-assured-groovy for some examples.
-         */
-        .body(
-            "identifier.find {it.system == 'https://bluebutton.cms.gov/resources/variables/clm_id'}.value",
-            equalTo(claimId))
-        .statusCode(200)
-        .when()
-        .get(eobEndpoint + eobId);
+  /**
+   * Verifies that an EOB HHA claim can be requested using the read endpoint, successfully returns a
+   * 200 response, and has the claim id and a couple other EOB details present in the body.
+   */
+  @Test
+  public void testReadHhaEobIdExpectClmIdInResponse() {
+    verifySuccessfulResponseAndClaimIdFor(ClaimType.HHA);
+  }
+
+  /**
+   * Verifies that an EOB Hospice claim can be requested using the read endpoint, successfully
+   * returns a 200 response, and has the claim id and a couple other EOB details present in the
+   * body.
+   */
+  @Test
+  public void testReadHospiceEobIdExpectClmIdInResponse() {
+    verifySuccessfulResponseAndClaimIdFor(ClaimType.HOSPICE);
+  }
+
+  /**
+   * Verifies that an EOB Inpatient claim can be requested using the read endpoint, successfully
+   * returns a 200 response, and has the claim id and a couple other EOB details present in the
+   * body.
+   */
+  @Test
+  public void testReadInpatientEobIdExpectClmIdInResponse() {
+    verifySuccessfulResponseAndClaimIdFor(ClaimType.INPATIENT);
+  }
+
+  /**
+   * Verifies that an EOB Outpatient claim can be requested using the read endpoint, successfully
+   * returns a 200 response, and has the claim id and a couple other EOB details present in the
+   * body.
+   */
+  @Test
+  public void testReadOutpatientEobIdExpectClmIdInResponse() {
+    verifySuccessfulResponseAndClaimIdFor(ClaimType.OUTPATIENT);
+  }
+
+  /**
+   * Verifies that an EOB PDE claim can be requested using the read endpoint, successfully returns a
+   * 200 response, and has the claim id and a couple other EOB details present in the body.
+   */
+  @Test
+  public void testReadPdeEobIdExpectClmIdInResponse() {
+    verifySuccessfulResponseAndClaimIdFor(ClaimType.PDE);
+  }
+
+  /**
+   * Verifies that an EOB SNF claim can be requested using the read endpoint, successfully returns a
+   * 200 response, and has the claim id and a couple other EOB details present in the body.
+   */
+  @Test
+  public void testReadSnfEobIdExpectClmIdInResponse() {
+    verifySuccessfulResponseAndClaimIdFor(ClaimType.SNF);
   }
 
   /** Verifies that the EOB returns a 400 when called with a negative count parameter. */
   @Test
   public void testEobByPatientIdWithNegativeCountExpect400() {
 
     String patientId = testUtils.getPatientId(testUtils.loadSampleAData());
-    // It's also possible to add these to the call below, but I liked the request string as
-    // one would write it when calling the API
+    // Its possible to add these via the API, but
```

Comment:
seems a little wonky....", but" what?

Reply:
Oops, had a fix for this and never pushed it. Fixed!